### DISCUSSION TITLE
Renovate not upgrade the Python package "black" from 20.8b1 to 21.5b1

### DISCUSSION THREAD

**Discussion by hongbo-miao** [renovatebot/renovate#10199] [Request Help] [answered]
It has been a while for new version. But Renovate not open pull request to upgrade the Python package [black](https://github.com/psf/black/) from 20.8b1 to 21.5b1
https://github.com/Hongbo-Miao/hongbomiao.com/blob/main/Pipfile#L12

I was thinking it may be related with unstable versions, however based on https://docs.renovatebot.com/configuration-options/#ignoreunstable

> By default, Renovate won't update any package versions to unstable versions (e.g. 4.0.0-rc3) unless the current version has the same major.minor.patch and was already unstable (e.g. it was already on 4.0.0-rc2).

If Renovate consider 20.8b1 as a  unstable version, then it still should upgrade to 21.5b1 as the current version is already unstable.

Any idea? Thanks

**Comment by rarkins** [comment, marked as answer]
Please start by looking at the debug log files and search for references to that dependency

**Reply by hongbo-miao** [reply]
Hmm any document for how to get debug log? I cannot find after searching "how to debug Renovate". Thanks

**Reply by hongbo-miao** [reply]
Oh I think you mean https://app.renovatebot.com/

**Reply by rarkins** [reply]
Yes, if you're using the hosted platform then the last 3 days of logs are available at app.renovatebot.com

**Reply by hongbo-miao** [reply]
Here you are. In latest log. This is only section I found has `black` inside.

```js
// ...
    "pipenv": [
      {
        "packageFile": "Pipfile",
        "deps": [
          {
            "depType": "dev-packages",
            "depName": "black",
            "managerData": {},
            "currentValue": "==20.8b1",
            "datasource": "pypi",
            "depIndex": 0,
            "warnings": [],
            "currentVersion": "20.8b1",
            "fixedVersion": "20.8b1",
            "updates": []
          },
```

**Reply by rarkins** [reply]
I think the issue is that:
- Every release is a "beta"/unstable release
- Renovate by default does not update pre-releases unless they are the same major.minor.patch

I think the following will work for this package:

```json
{
  "packageRules": [{
    "matchDatasources": ["pypi"],
    "matchPackageNames": ["black"],
    "ignoreUnstable": false
  }]
}
```

**Reply by hongbo-miao** [reply]
Thanks!
